Sources/Networking/CustomerAPI.swift

```diff
@@ -109,16 +104,14 @@ final class CustomerAPI {
         }
 
         let config = NetworkOperation.UserSpecificConfiguration(httpClient: self.backendConfig.httpClient,
-                                                                appUserID: appUserID)
+                                                                appUserID: transactionData.appUserID)
 
-        let postData = PostReceiptDataOperation.PostData(appUserID: appUserID,
-                                                         receiptData: receiptData,
-                                                         isRestore: isRestore,
-                                                         productData: productData,
-                                                         presentedOfferingIdentifier: offeringIdentifier,
-                                                         observerMode: observerMode,
-                                                         initiationSource: initiationSource,
-                                                         subscriberAttributesByKey: subscriberAttributesToPost)
+        let postData = PostReceiptDataOperation.PostData(
+            transactionData: transactionData.withAttributesToPost(subscriberAttributesToPost),
```

Comment:
So we are recreating the transaction data here since the `consentStatus` could be added to the attributes correct? If so, I wonder if we could extract that piece of code adding that attribute to the moment where we create the `PurchasedTransactionData` to avoid this recreation. Not a blocker though.